packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -707,6 +691,42 @@ class ZoomPageTransitionsBuilder extends PageTransitionsBuilder {
   // for the Impeller backend.
   static const bool _kProfileForceDisableSnapshotting = bool.fromEnvironment('flutter.benchmarks.force_disable_snapshot');
 
+  @override
+  DelegatedTransitionBuilder? get delegatedTransitionBuilder => (BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget? child) {
+    return delegateTransition(context, animation, secondaryAnimation, child, allowSnapshotting, allowEnterRouteSnapshotting);
+  };
+
+  /// The delegated transition.
+  static Widget delegateTransition(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget? child, bool allowSnapshotting, bool allowEnterRouteSnapshotting) {
+    return DualTransitionBuilder(
+      animation: ReverseAnimation(secondaryAnimation),
+      forwardBuilder: (
+        BuildContext context,
+        Animation<double> animation,
+        Widget? child,
+      ) {
+        return _ZoomEnterTransition(
+          animation: animation,
+          allowSnapshotting: allowSnapshotting && allowEnterRouteSnapshotting,
+          reverse: true,
+          child: child,
+        );
+      },
+      reverseBuilder: (
+        BuildContext context,
+        Animation<double> animation,
+        Widget? child,
+      ) {
+        return _ZoomExitTransition(
+          animation: animation,
+          allowSnapshotting: allowSnapshotting,
+          child: child,
+        );
+      },
+      child: child,
+    );
```

Comment:
Maybe we discussed this before, but is this block duplicated with the non-delegated ZoomPageTransition?